plugins/hypervisors/vmware/src/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -1363,24 +1364,15 @@ private Answer attachVolume(Command cmd, DiskTO disk, boolean isAttach, boolean
             AttachAnswer answer = new AttachAnswer(disk);
 
             if (isAttach) {
-                String dataDiskController = controllerInfo.get(VmDetailConstants.DATA_DISK_CONTROLLER);
-                String rootDiskController = controllerInfo.get(VmDetailConstants.ROOK_DISK_CONTROLLER);
-                DiskControllerType rootDiskControllerType = DiskControllerType.getType(rootDiskController);
-
-                if (dataDiskController == null) {
-                    dataDiskController = getLegacyVmDataDiskController();
-                } else if ((rootDiskControllerType == DiskControllerType.lsilogic) ||
-                           (rootDiskControllerType == DiskControllerType.lsisas1068) ||
-                           (rootDiskControllerType == DiskControllerType.pvscsi) ||
-                           (rootDiskControllerType == DiskControllerType.buslogic)) {
-                    //TODO: Support mix of SCSI controller types for single VM. If root disk is already over
-                    //a SCSI controller then use the same for data volume as well. This limitation will go once mix
-                    //of SCSI controller types for single VM.
-                    dataDiskController = rootDiskController;
-                } else if (DiskControllerType.getType(dataDiskController) == DiskControllerType.osdefault) {
-                    dataDiskController = vmMo.getRecommendedDiskController(null);
+                String diskController = getLegacyVmDataDiskController();
```

Comment:
This has been merged into `4.7` and have forward merged into `4.8` without a problem.

I have merge conflicts trying to merge this from `4.8` into `master` though.

Here is the merge conflict:
```
            AttachAnswer answer = new AttachAnswer(disk);

            if (isAttach) {
<<<<<<< HEAD
                String dataDiskController = controllerInfo.get(VmDetailConstants.DATA_DISK_CONTROLLER);
                String rootDiskController = controllerInfo.get(VmDetailConstants.ROOT_DISK_CONTROLLER);
                DiskControllerType rootDiskControllerType = DiskControllerType.getType(rootDiskController);

                if (dataDiskController == null) {
                    dataDiskController = getLegacyVmDataDiskController();
                } else if ((rootDiskControllerType == DiskControllerType.lsilogic) ||
                           (rootDiskControllerType == DiskControllerType.lsisas1068) ||
                           (rootDiskControllerType == DiskControllerType.pvscsi) ||
                           (rootDiskControllerType == DiskControllerType.buslogic)) {
                    //TODO: Support mix of SCSI controller types for single VM. If root disk is already over
                    //a SCSI controller then use the same for data volume as well. This limitation will go once mix
                    //of SCSI controller types for single VM.
                    dataDiskController = rootDiskController;
                } else if (DiskControllerType.getType(dataDiskController) == DiskControllerType.osdefault) {
                    dataDiskController = vmMo.getRecommendedDiskController(null);
=======
                String diskController = getLegacyVmDataDiskController();
                if (controllerInfo != null &&
                        !Strings.isNullOrEmpty(controllerInfo.get(VmDetailConstants.DATA_DISK_CONTROLLER))) {
                    diskController = controllerInfo.get(VmDetailConstants.DATA_DISK_CONTROLLER);
>>>>>>> 4.8
                }
                if (DiskControllerType.getType(diskController) == DiskControllerType.osdefault) {
                    diskController = vmMo.getRecommendedDiskController(null);
                }
                vmMo.attachDisk(new String[] {datastoreVolumePath}, morDs, diskController);
            } else {
```

Please advise...